src/wasm/wat-lexer.cpp

```diff
@@ -781,6 +781,136 @@ std::optional<LexResult> keyword(std::string_view in) {
 
 } // anonymous namespace
 
+std::optional<uint64_t> Token::getU64() const {
+  if (auto* tok = std::get_if<IntTok>(&data)) {
+    if (tok->sign == NoSign) {
+      return tok->n;
+    }
+  }
+  return {};
+}
+
+std::optional<int64_t> Token::getS64() const {
+  if (auto* tok = std::get_if<IntTok>(&data)) {
+    // Integers without an explicit sign still need to be in signed range.
+    if (tok->sign != NoSign || tok->n < (1ull << 63)) {
```

Comment:
I think we have several more places to check the limit like this. Should we fix all of them?

Reply:
This would actually have to be `uint64_t(std::numeric_limits<int64_t>::min())`. Between the verbosity and the cast, the bit twiddling seemed nicer, but if we'd prefer to use `numeric_limits` instead, I can go fix it up everywhere. What would you two prefer?

Reply:
It looks like we use `numeric_limits` in more places than we write out the constant, from some quick grepping. So I'd lean in that direction. But fair point on the verbosity, I don't feel strongly.

Reply:
I'd also lean towards `numeric_limits` too, given that it's self-explanatory.

Reply:
What would you think about using `INT64_MIN` and friends as a less verbose but similarly symbolic option?

Reply:
`INT64_MIN` sgtm.